pingora-core/src/protocols/ssl/client.rs

```diff
@@ -43,13 +43,30 @@ pub async fn handshake<S: IO>(
         Err(e) => {
             let context = format!("TLS connect() failed: {e}, SNI: {domain}");
             match e.code() {
-                ssl::ErrorCode::SSL => match stream.ssl().verify_result().as_raw() {
-                    // X509_V_ERR_INVALID_CALL in case verify result was never set
-                    X509_V_OK | X509_V_ERR_INVALID_CALL => {
-                        Error::e_explain(TLSHandshakeFailure, context)
+                ssl::ErrorCode::SSL => {
+                    #[cfg(not(feature = "boringssl"))]
```

Comment:
Thanks for the submission something does look wrong here. This code looks good, it appears `X509_V_OK` is [equivalent](https://docs.rs/boring/latest/src/boring/x509/mod.rs.html#1412) to `Ok(())` so I think the `X509_V_OK` match on the `boringssl` block is unnecessary.

I think we could clean this up a bit if we unified the type we're matching against, something like this.

  ```
    #[cfg(not(feature = "boringssl"))]
    fn verify_result<S>(stream: SslStream<S>) -> Result<(), i32> {
        match stream.ssl().verify_result().as_raw() {
            X509_V_OK => Ok(()),
            e => Err(e),
        }
    }

    #[cfg(feature = "boringssl")]
    fn verify_result<S>(stream: SslStream<S>) -> Result<(), i32> {
        stream.ssl().verify_result().map_err(|e| e.as_raw())
    }

    ....

    match verify_result(stream) {
        Ok(()) => Error::e_explain(TLSHandshakeFailure, context),
        // X509_V_ERR_INVALID_CALL in case verify result was never set
        Err(X509_V_ERR_INVALID_CALL) => {
            Error::e_explain(TLSHandshakeFailure, context)
        }
        _ => Error::e_explain(InvalidCert, context),
    }
``` 
```

Reply:
Thanks for your review!
Indeed, it would be more elegant to write this way. I have modified the corresponding code.